frontend/src/components/MapView/Layers/AnticipatoryActionLayer/index.tsx

```diff
@@ -5,61 +5,123 @@ import { useSelector } from 'react-redux';
-import { AnticipatoryActionDataSelector } from 'context/anticipatoryActionStateSlice';
+import {
+  AACategoryFiltersSelector,
+  AASelectedWindowSelector,
+  AnticipatoryActionDataSelector,
+  allWindowsKey,
+} from 'context/anticipatoryActionStateSlice';
 import { Layer, Source } from 'react-map-gl/maplibre';
 import { getFormattedDate } from 'utils/date-utils';
 import { DateFormat } from 'utils/name-utils';
+import {
+  AADataSeverityOrder,
+  getAAColor,
+} from 'components/MapView/LeftPanel/AnticipatoryActionPanel/utils';
 
 const boundaryLayer = getBoundaryLayerSingleton();
 
 function AnticipatoryActionLayer({ layer, before }: LayersProps) {
-  // TODO: selectedDate instead of URL
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const selectedDate = useDefaultDate(layer.id);
   const AAData = useSelector(AnticipatoryActionDataSelector);
+  const aaWindow = useSelector(AASelectedWindowSelector);
+  const aaCategories = useSelector(AACategoryFiltersSelector);
   const boundaryLayerState = useSelector(
     layerDataSelector(boundaryLayer.id),
   ) as LayerData<BoundaryLayerProps> | undefined;
   const { data } = boundaryLayerState || {};
 
   const date = getFormattedDate(selectedDate, DateFormat.Default);
 
-  const adminToDraw = Object.entries(AAData)
-    .filter(x => x[1].find(y => y.date === date))
-    .map(x => x[0]);
-  const filteredData = data && {
-    ...data,
-    features: data.features.filter(cell =>
-      adminToDraw.includes(
-        cell.properties?.[boundaryLayer.adminLevelLocalNames[1] as string],
+  const districtsWithColors = React.useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(AAData).map(([district, districtData]) => {
+          const sameDateAndWindow = districtData.filter(
+            x =>
+              x.date === date &&
+              (aaWindow === allWindowsKey || x.windows === aaWindow),
+          );
+
+          if (sameDateAndWindow.length === 0) {
+            return [district, getAAColor('ny', 'ny', true)];
+          }
+
+          const active = sameDateAndWindow.filter(
+            x =>
+              x.probability !== 'NA' &&
+              Number(x.probability) >= Number(x.trigger),
+          );
+          if (active.length === 0) {
+            return [district, getAAColor('na', 'na', true)];
+          }
+
+          const max = active.reduce(
+            (prev, curr) => {
+              const currVal = AADataSeverityOrder(curr.category, curr.phase);
+              return currVal > prev.val ? { val: currVal, data: curr } : prev;
+            },
+            { val: -1, data: active[0] },
+          );
+
+          if (aaCategories[max.data.category] === false) {
+            return [district, null];
+          }
+
+          return [
+            district,
+            getAAColor(max.data.category, max.data.phase, true),
+          ];
+        }),
       ),
-    ),
-  };
+    [AAData, aaCategories, aaWindow, date],
+  );
```

Comment:
Let's find a way to share a utility or access the processed data here already so that we can control the logic in one place vs duplicating it.